ion-schema/src/constraint.rs

```diff
@@ -218,132 +219,170 @@ impl Constraint {
         Ok(Constraint::Regex(regex.try_into()?))
     }
 
+    /// Resolves all ISL type references to corresponding [TypeId]s
+    fn resolve_type_references_to_type_ids(
```

Comment:
Previously each constraint had its own method to parse type ids. This is just a reusable method that can be used in multiple places.